src/client.coffee

```diff
@@ -77,8 +78,7 @@ class Client
     if secret?
       @requestOptions.headers["Authorization"] = secret
 
-    # This works even if no apiKey passed but passed an "Authorization" header
-    if "Authorization" of @requestOptions.headers
+    if address.search(/doofinder.com/) != -1
```

Comment:
Why this change?

Reply:
I need to disable https and SSL security in local development because with this it didin't work.
So the unique way I found is to check if the doofinder.com is in the target address, check if the authentication token is defined is not longer valid as we are gonna use it in local enviroment also and playground for example wouldn't work.

Reply:
🤔 maybe this could help… starting refactoring at line 75…

```coffeescript
if secret?
  @requestOptions.headers["Authorization"] = secret

if protocol?
  @requestOptions.protocol = "#{protocol}:"
elif "Authorization" of @requestOptions.headers
  @requestOptions.protocol = "https:"
  forceSSL = true
```

I haven't tried it but I think that this would allow you to force the protocol even providing the auth token.